onnxruntime/core/providers/nnapi/nnapi_builtin/builders/op_builder.cc

```diff
@@ -89,27 +105,30 @@ void AddTransposeOperator(ModelBuilder& model_builder,
 
   Shape perm_dimen = {SafeInt<uint32_t>(perm.size())};
   OperandType perm_operand_type(Type::TENSOR_INT32, perm_dimen);
-  uint32_t perm_idx = model_builder.AddOperandFromPersistMemoryBuffer(
-      perm_name, perm.data(), perm_operand_type);
+  ORT_RETURN_IF_ERROR(model_builder.AddOperandFromPersistMemoryBuffer(perm_name, perm.data(), perm_operand_type));
+  uint32_t perm_idx = operand_indices.at(perm_name);
 
   input_indices.push_back(perm_idx);  // permutation
-  shaper.Transpose(input, perm, output);
+  ORT_RETURN_IF_ERROR(shaper.Transpose(input, perm, output));
   OperandType output_operand_type = operand_types.at(input);
   output_operand_type.SetDimensions(shaper[output]);
-  model_builder.AddOperation(ANEURALNETWORKS_TRANSPOSE, input_indices, {output},
-                             {output_operand_type}, {output_is_nhwc});
-}
-
-void TransposeBetweenNCHWAndNHWC(ModelBuilder& model_builder,
-                                 const std::string& input,
-                                 const std::string& output,
-                                 bool nchw_to_nhwc) {
-  ORT_ENFORCE(!model_builder.UseNCHW(), "model_builder.UseNCHW() is on");
+  return model_builder.AddOperation(ANEURALNETWORKS_TRANSPOSE, input_indices, {output},
+                                    {output_operand_type}, {output_is_nhwc});
+}
+
+Status TransposeBetweenNCHWAndNHWC(ModelBuilder& model_builder,
+                                   const std::string& input,
+                                   const std::string& output,
+                                   bool nchw_to_nhwc) ORT_MUST_USE_RESULT;
+Status TransposeBetweenNCHWAndNHWC(ModelBuilder& model_builder,
+                                   const std::string& input,
+                                   const std::string& output,
+                                   bool nchw_to_nhwc) {
+  ORT_RETURN_IF_NOT(!model_builder.UseNCHW(), "model_builder.UseNCHW() is on");
   const auto& shaper(model_builder.GetShaper());
-  ORT_ENFORCE(
-      4 == shaper[input].size(),
-      "TransposeNCHWToNHWC input has to be a 4d tensor, actual dimensions: " +
-          std::to_string(shaper[input].size()));
+  ORT_RETURN_IF_NOT(4 == shaper[input].size(),
+                    "TransposeNCHWToNHWC input has to be a 4d tensor, actual dimensions: ",
+                    std::to_string(shaper[input].size()));
 
```

Comment:
fwiw you shouldn't need the std::to_string call when using the MakeString functionality. not worth another round of builds to change though

Reply:
Will correct in my other PRs

---
In reply to: [465459760](https://github.com/microsoft/onnxruntime/pull/4701#discussion_r465459760) [](ancestors = 465459760)